javascript/ql/src/Security/CWE-300/InsecureDependencyResolution.qhelp

```diff
@@ -0,0 +1,55 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+
+<overview>
+<p>
+Using an insecure protocol like HTTP or FTP to download build dependencies makes the build process vulnerable to a 
+Man in the Middle (MITM) attack.
+</p>
+<p>
+This can allow attackers to inject malicious code into the downloaded dependencies and thereby 
+infect the build artifacts and execute arbitrary code on the machine building the artifacts.
+</p>
+
+</overview>
+<recommendation>
+
+<p>Always use HTTPS or SFTP when downloading artifacts from an URL.</p>
+
+</recommendation>
+
+<example>
+<p>
+The below example shows a <code>package.json</code> file that downloads a dependency using unencrypted HTTP.
```

Comment:
```suggestion
The below example shows a <code>package.json</code> file that downloads a dependency using the insecure HTTP protocol.
```
I think it is preferable to talk about the vaguer secure/insecure instead of encrypted. Technically, you just need the packets from the server to be signed, it is fine if they are transmitted in cleartext.

... which leads me to consider if we should have a line about verifying the md5sum of insecurely downloaded artifacts as an alternative to the secure download.
I do not think we need it here though, that can be added for a overly aggressive query that checks if any md5sum verification is done on any downloaded resources.

Reply:
> which leads me to consider if we should have a line about verifying the md5sum of insecurely downloaded artifacts as an alternative to the secure download. 

Can you do that for dependencies downloaded through a `package.json` file? 
Wouldn't that require downloading the dependency (without running install script), verifying a checksum, and then run the install script.   
I think such a setup is sufficiently rare that we don't need to mention it.  

It's common on `<script/>` tags, but that's another matter.

Reply:
> `It's common on <script/> tags, but that's another matter.`

Great idea: https://github.com/github/codeql-javascript-team/issues/374